src/menu/menu.js

```diff
@@ -1,81 +1,136 @@
+    <p id="latitude" style="display: inline-block;"></p>
+    <br>
+    <h4 id="longitude-label" style="display: inline-block;">Longitude:</h4>
+    <p id="longitude" style="display: inline-block;"></p>
+    <h1>Menu</h1>
+  </div>
+ `;
+}
+
 export function getMenuTemplate() {
-  return ` <h1>Menu</h1>`;
+  return `
+  <table>
+  <tbody id="main"></tbody>
+  <tfoot>
+    <tr id="total-row-quantity">
+      <td colspan="2" id="total-quantity">Total quantity: 0</td>
+    </tr>
+    <br>
+    <tr id="total-row-price">
+      <td colspan="2" id="total-price">Total price: $0.00</td>
+    </tr>
+  </tfoot>
+  </table>
+  `;
+}
+
+export function getRestaurantDetailsByID(id) {
+  fetchData(
+    `https://private-anon-7231255228-pizzaapp.apiary-mock.com/restaurants/${id}`
+  )
+    .then((result) => {
+      document.getElementById("name").innerHTML = result.name;
+      document.getElementById("address1").innerHTML = result.address1;
+      document.getElementById("address2").innerHTML = result.address2;
+      document.getElementById("latitude").innerHTML = result.latitude;
+      document.getElementById("longitude").innerHTML = result.longitude;
+      getRestaurantMenu();
+    })
+    .catch((error) => {
+      console.error(error);
+    });
 }
 
 export function getRestaurantMenu(id) {
   fetchData(
     `https://private-anon-7231255228-pizzaapp.apiary-mock.com/restaurants/${id}/menu?category=Pizza&orderBy=rank`
-  ).then((result) => {
-    const menuBody = document.getElementById("main");
-
-    let totalPrice = 0;
-    let totalQuantity = 0;
-
-    result.forEach((item) => {
-      const row = document.createElement("tr");
-      const name = document.createElement("td");
-      const topping = document.createElement("td");
-      const price = document.createElement("td");
-      const addToCartBtn = document.createElement("button");
-
-      name.innerText = item.name;
-
-      topping.innerText = item.topping?.join(", ") || " ";
-      price.innerText = ` $ ${item.price}`;
-      addToCartBtn.innerText = "Add to cart";
-
-      addToCartBtn.addEventListener("click", () => {
-        let cart = JSON.parse(localStorage.getItem("cart"));
-        if (!Array.isArray(cart)) {
-          cart = [];
-        }
-
-        //get id from URL
-        const id = window.location.hash.split("/")[1];
-        console.log("the id is ", id);
-
-        const menuItem = { item, id };
-        cart.push(menuItem);
-        localStorage.setItem("cart", JSON.stringify(cart));
-
-        totalQuantity++;
-        totalPrice += item.price;
-
-        alert(`${item.name} has been added to your cart!`);
-        console.log(menuItem);
-
-        // Update total row
-        totalQuantityCol.innerText = `Total quantity: ${totalQuantity}`;
-        totalPriceCol.innerText = `Total price: $${totalPrice.toFixed(2)}`;
+  )
+    .then((result) => {
+      const menuBody = document.getElementById("main");
+      document.getElementById("loader")?.remove();
```

Comment:
Here the loader is removed but there's still data to be added so you should move it at the end of the function.
After removing the loader you should also remove the class that hides the main content if you go with this approach.